types/eslint/index.d.ts

```diff
@@ -34,6 +34,31 @@ export namespace AST {
         loc: SourceLocation;
     }
 
+    interface PunctuatorToken<Value extends string> extends Token {
+        type: 'Punctuator';
+        value: Value;
+    }
+
+    interface ArrowToken extends PunctuatorToken<'=>'> {}
+
+    interface CommaToken extends PunctuatorToken<','> {}
+
+    interface SemicolonToken extends PunctuatorToken<';'> {}
+
+    interface ColonToken extends PunctuatorToken<':'> {}
+
+    interface OpeningParenToken extends PunctuatorToken<'('> {}
+
+    interface ClosingParenToken extends PunctuatorToken<')'> {}
+
+    interface OpeningBracketToken extends PunctuatorToken<'['> {}
+
+    interface ClosingBracketToken extends PunctuatorToken<']'> {}
+
+    interface OpeningBraceToken extends PunctuatorToken<'{'> {}
+
+    interface ClosingBraceToken extends PunctuatorToken<'}'> {}
```

Comment:
I'm wondering if these should be defined in `types/estree` instead... I originally interpreted the request to place them in `types/eslint`, but happy to move if someone thinks they should be elsewhere!

Reply:
Tokens aren't part of the ESTree spec - so it makes sense that they are not declared in the `@types/estree` package.

Reply:
Swell, I'll move them into `eslint-utils`. 👍